lite/valor_lite/detection/manager.py

```diff
@@ -557,132 +559,321 @@ def _compute_detailed_counts(
             List of IoU thresholds to compute metrics for.
         score_thresholds : list[float], default=[0.1,0.2,...,1.0]
             List of confidence thresholds to compute metrics for.
-        n_samples : int, default=0
-            Number of datum samples to return per metric.
+        number_of_examples : int, default=0
+            Maximum number of annotation examples to return per metric.
 
         Returns
         -------
-        list[list[DetailedCounts]]
+        list[list[ConfusionMatrix]]
             Outer list is indexed by label, inner list is by IoU.
         """
 
         if self._detailed_pairs.size == 0:
             return list()
 
-        metrics = compute_detailed_counts(
-            data=self._detailed_pairs,
+        (
+            confusion_matrix,
+            hallucinations,
+            missing_predictions,
+        ) = compute_confusion_matrix(
+            self._detailed_pairs,
             label_metadata=self._label_metadata,
             iou_thresholds=np.array(iou_thresholds),
             score_thresholds=np.array(score_thresholds),
-            n_samples=n_samples,
+            n_examples=number_of_examples,
         )
 
-        tp_idx = 0
-        fp_misclf_idx = 2 * n_samples + 1
-        fp_halluc_idx = 4 * n_samples + 2
-        fn_misclf_idx = 6 * n_samples + 3
-        fn_misprd_idx = 8 * n_samples + 4
-
-        def _unpack_examples(
-            iou_idx: int,
-            label_idx: int,
-            type_idx: int,
-            example_source: dict[int, NDArray[np.float16]],
-        ) -> list[list[tuple[str, tuple[float, float, float, float]]]]:
+        def _unpack_confusion_matrix(
```

Comment:
- move helper out of function so it isn't reinitialized when function is called?
- nit: add one-liner docstring?